psi4/driver/driver_findif.py

```diff
@@ -26,119 +26,154 @@
 # @END LICENSE
 #
 
+import copy
+import logging
+from functools import partial
+from typing import Any, Callable, Dict, Iterator, List, Tuple, Union
+
 import numpy as np
+import pydantic
+from qcelemental.models import DriverEnum, AtomicResult
+from qcelemental import constants
+
 from psi4 import core
-from psi4.driver import p4util
+from psi4.driver import p4util, pp, qcdb, nppp10
 from psi4.driver.p4util.exceptions import ValidationError
-from psi4.driver import qcdb
+
+logger = logging.getLogger(__name__)
 
 # CONVENTIONS:
 # n_ at the start of a variable name is short for "number of."
 # _pi at the end of a variable name is short for "per irrep."
 # h is the index of an irrep.
 
-array_format = {"precision": 10}
-
 
-def _displace_cart(mol, geom, salc_list, i_m, step_size):
+def _displace_cart(mass: np.ndarray, geom: np.ndarray, salc_list: "psi4.core.CdSalcList", i_m: Iterator[Tuple], step_size: float) -> Tuple[np.ndarray, str]:
```

Comment:
Are the quotes around `psi4.core.CdSalcList` needed?

Reply:
quotes are used when you want to specify the type but actually importing it and having the object available would cause trouble. here, unquoted `core.CdSalcList` is probably safe. good catch.

I figure I'm going to be adjusting the signature and docstrings formatting someday when I can look at the built docs.